### ISSUE TITLE
[UX] admin menu: standardize to "List xyz" instead of plain "List"

### ISSUE THREAD

**Issue by klonos** [backdrop/backdrop-issues#525]
So, we have:

```Structure``` -> ```Content types``` -> ```List```
```Structure``` -> ```Taxonomy``` -> ```List```
```Structure``` -> ```Views``` -> ```List```
```Configuration``` -> ```Content authoring``` -> ```Text formats``` -> ```List```
```Configuration``` -> ```Media``` -> ```Image styles``` -> ```List```
```Configuration``` -> ```Search and metadata``` -> ```URL aliases``` -> ```List```
```Reports``` -> ```Available updates``` -> ```List```
```Reports``` -> ```Field list``` -> ```List```

but:

```Structure``` -> ```Custom blocks``` -> ```List blocks```
```Structure``` -> ```Layouts``` -> ```List layouts```
```Structure``` -> ```Menus``` -> ```List menus```

I suggest standardizing to "List xyz" instead of plain "List" because (besides my OCD kicking in):

![list_xyz](https://cloud.githubusercontent.com/assets/2423362/5591565/c73658e2-91bd-11e4-88ee-96a480e7580a.png)

After #947 we now have another group of "List" menu entries added to the existing set.

I think we should have this instead:

- [x] ```Structure``` -> ```Content types``` -> ```List content types``` (currently plain "List")
- [x] ```Structure``` -> ```Custom blocks``` -> ```List blocks``` (ok as is)
- [x] ```Structure``` -> ```Layouts``` -> ```List layouts``` (ok as is)
- [x] ```Structure``` -> ```Menus``` -> ```List menus``` (ok as is)
- [x] ```Structure``` -> ```Taxonomy``` -> ```List vocabularies``` (currently plain "List")
- [x] ```Structure``` -> ```Taxonomy``` -> ```[vocabulary_name]``` -> ```List terms``` (currently plain "List")
- [x] ```Structure``` -> ```Views``` -> ```List views``` (currently plain "List")
- [x] ```Configuration``` -> ```Content authoring``` -> ```Text formats``` -> ```List text formats``` (currently plain "List")
- [x] ```Configuration``` -> ```Development``` -> ```Testing``` -> ```List tests``` (currently plain "List")
- [x] ```Configuration``` -> ```Media``` -> ```Image styles``` -> ```List image styles``` (currently plain "List")
- [x] ```Configuration``` -> ```Regional and language``` -> ```Languages``` -> ```List languages``` (currently plain "List")
- [x] ```Configuration``` -> ```Search and metadata``` -> ```URL aliases``` -> ```List URL aliases``` (currently plain "List")
- [x] ```Configuration``` -> ```Search and metadata``` -> ```URL aliases``` -> ```Patterns``` -> ```List patterns``` (currently "Pattern List")
- [x] ```Reports``` -> ```Available updates``` -> ```List available updates``` (currently plain "List")
- [x] ```Reports``` -> ```Fields``` -> ```List fields``` (currently plain ```Reports``` -> ```Field list``` -> ```List```)

**Comment by klonos** [comment]
...minor notes:

```Reports``` -> ```Field list``` -> ```List``` should change to ```Reports``` -> ```Fields``` -> ```List fields```

...because:

![fields_list](https://cloud.githubusercontent.com/assets/2423362/5591668/44b203d2-91ca-11e4-8341-5f97fe05f879.gif)

```Structure``` -> ```Custom blocks``` -> ```List blocks``` should be ```Structure``` -> ```Custom blocks``` -> ```List custom blocks``` (either that or ```Structure``` -> ```Blocks``` -> ```List blocks```)

...for consistency.

Also, similar operation links are listed in the suggestions of the search in the admin bar as ```[top-level-menu]: [operation]```. For example, searching for "add" shows among the rest of the suggestions 3 "Structure: Add link" results. These are the respective "Add link" menu items for the 3 menus that come with backdrop out of the box. This is just an example among quite a few others where it doesn't make sense to simply list these as ```[top-level-menu]: [operation]``` without also giving a hint about the respective menu. Is there any way to either be more specific for these like 
```[top-level-menu]: [parent]: [operation]```. If so, in the example I mentioned, the 3 similar "Structure: Add link" search results would instead be:

"Structure: Account menu: Add link"
"Structure: Administration menu: Add link"
"Structure: Primary navigation: Add link"

I realize that this last one is really complicated, so we can leave it for later on.

**Comment by klonos** [comment]
This issue is similar to #399

**Comment by jenlampton** [comment]
yes please :) +1

**Comment by quicksketch** [comment]
+1 Great idea.

**Comment by klonos** [comment]
I was thinking... what is the point of having for example *Structure > Content types > List* (and the respective ```/admin/structure/types/list``` path) when you get to the same page from *Structure > Content types* (/admin/structure/types)? Same goes for pretty much every other menu item under *Structure > * and elsewhere where we have "List [something]" menu items.

Should we just get rid of all "List" items and be done with it? I'm asking because I'm trying to see where we are with #523 and this issue here is a great part of it.

**Comment by BWPanda** [comment]
@klonos Only reason I can think is because some people don't realise you can click on a parent link if, when you hover over it, a sub-menu appears...

**Comment by klonos** [comment]
Fair enough then, that sounds like a good reason. Back to the original plan of having "List xyz" instead of just "List".

**Comment by klonos** [comment]
After #947 we now have another group of "List" menu entries added to the existing set.

I think we should have this instead:

```Structure``` -> ```Content types``` -> ```List content types``` (currently plain "List")
```Structure``` -> ```Custom blocks``` -> ```List blocks``` (ok as is)
```Structure``` -> ```Layouts``` -> ```List layouts``` (ok as is)
```Structure``` -> ```Menus``` -> ```List menus``` (ok as is)
```Structure``` -> ```Taxonomy``` -> ```List vocabularies``` (currently plain "List")
```Structure``` -> ```Taxonomy``` -> ```[vocabulary_name]``` -> ```List terms``` (currently plain "List")
```Structure``` -> ```Views``` -> ```List views``` (currently plain "List")
```Configuration``` -> ```Content authoring``` -> ```Text formats``` -> ```List text formats``` (currently plain "List")
```Configuration``` -> ```Media``` -> ```Image styles``` -> ```List image styles``` (currently plain "List")
```Configuration``` -> ```Regional and language``` -> ```Languages``` -> ```List languages``` (currently plain "List")
```Configuration``` -> ```Search and metadata``` -> ```URL aliases``` -> ```List URL aliases``` (currently plain "List")
```Configuration``` -> ```Search and metadata``` -> ```URL aliases``` -> ```Patterns``` -> ```List patterns``` (currently "Pattern List")
```Reports``` -> ```Available updates``` -> ```List available updates``` (currently plain "List")
```Reports``` -> ```Fields``` -> ```List fields``` (currently plain ```Reports``` -> ```Field list``` -> ```List```)

**Comment by klonos** [comment]
...Also, even if we implemented the above, we'd be 90% there. You see there would be these menu items:

```Structure``` -> ```Taxonomy``` -> ```Tags``` -> ```List terms```
```Structure``` -> ```Taxonomy``` -> ```Categories``` -> ```List terms```
```Structure``` -> ```Taxonomy``` -> ```Another vocabulary``` -> ```List terms```

...as well as these ones:

```Structure``` -> ```Menus``` -> ```Account menu``` -> ```List links```
```Structure``` -> ```Menus``` -> ```Administration menu``` -> ```List links```
```Structure``` -> ```Menus``` -> ```Primary navigation``` -> ```List links```
```Structure``` -> ```Menus``` -> ```Another menu``` -> ```List links```

So if one was to enter the search term "list" in the admin bar search field, they'd get this:

...
Structure: List terms
Structure: List terms
Structure: List terms
...
Structure: List links
Structure: List links
Structure: List links
Structure: List links
...

This is way better than plain plain "Structure: List" for all these results, but still... can we (I don't know if this is possible) have it so that the search results (*just* the search results - not the menu items themselves) are like so instead:

...
Structure: List *Tags* terms
Structure: List *Categories* terms
Structure: List *Another vocabulary* terms
...
Structure: List *Account menu* links
Structure: List *Administration menu* links
Structure: List *Primary navigation* links
Structure: List *Another menu* links
...

**Comment by klonos** [comment]
...or like so perhaps (prefix the "immediate" parent menu item along with the top-most parent item):

...
Structure: Tags: List terms
Structure: Categories: List terms
Structure: Another vocabulary: List terms
...
Structure: Account menu: List links
Structure: Administration menu: List links
Structure: Primary navigation: List links
Structure: Another menu: List links
...

**Comment by quicksketch** [comment]
> This is way better than plain plain "Structure: List" for all these results, but still... can we (I don't know if this is possible) have it so that the search results (just the search results - not the menu items themselves) are like so instead:

Yeah, let's do that issue separately, it would probably require a different solution. As-is, the current problem mostly just involves updating the entries in hook_menu().

**Comment by klonos** [comment]
> Yeah, let's do that issue separately...

Done.

**Comment by klonos** [comment]
Nice! We're a step closer to perfection!!: Here's a screenie of how the search menu looks like now when searching the term "list" (with #1001 applied to increase the search results width):

![admin_bar-search_results_for_term_list](https://cloud.githubusercontent.com/assets/2423362/8222412/4ed529c6-15ae-11e5-9219-d9cfd9ac5cdb.png)

The "List languages" menu was missed by @hellomrcat, but I'm sure it'll be taken care of soon. So we only need to solve #1002 and that'd be it. Thanx!

**Comment by jenlampton** [comment]
New PR over at https://github.com/backdrop/backdrop/pull/1161

**Comment by klonos** [comment]
Here's a before/after screenshot (with #1001 applied):

![backdrop-list_xyz](https://cloud.githubusercontent.com/assets/2423362/12077908/76c903f2-b24e-11e5-8a95-8984fea590b9.png)

Looks good to me. Thanx @jenlampton

**Comment by quicksketch** [comment]
The PR at https://github.com/backdrop/backdrop/issues/1161 has some side-effects.

Pages that have tabs change their "List" tab to "List X" as well. Which IMO seems unnecessary and is probably unintentional:

![selection_028](https://cloud.githubusercontent.com/assets/100206/12094930/659442a0-b2c0-11e5-94a3-cdeedbe88fdb.png)
![selection_029](https://cloud.githubusercontent.com/assets/100206/12094934/6b37841a-b2c0-11e5-81db-e0bd2a610401.png)

Lots of other pages don't have tabs, and so the change doesn't have any other effects.

**Comment by jenlampton** [comment]
@quicksketch the change was unintentional, and yes, it's probably unnecessary, but I do think it's still an improvement. I think I'd rather have a clearer UI than a bunch of additional code to add different names for tabs than menu links.

**Comment by jenlampton** [comment]
Changing tag back to needs review for another look.  What do others think about the new accidental tab naming? Better or worse? 
@klonos @docwilmot @serundeputy ?

**Comment by klonos** [comment]
I don't mind it. In fact I like it better I think...

Previously, if one was reading the breadcrumb, then the page title and then the active tab in order to get a feeling where they are, it was like so:

Home > Administration > Structure > Menus > Menus

...it seemed like the page title was duplicated, but now:

Home > Administration > Structure > Menus > List menus

Better IMO.

This sort of solves an issue we still have with "Add x": #1507

**Comment by quicksketch** [comment]
Previously the menu item was "List", so the breadcrumb was:

Home > Administration > Structure > Menus > List

I'm not actually opposed to this change, I just want to evaluate it because it was unintentional.

**Comment by jenlampton** [comment]
I think everyone is in favor, let's move forward with this :)

**Comment by quicksketch** [comment]
Alright, merged in https://github.com/backdrop/backdrop/pull/1161 into 1.x for 1.3.0.

**Comment by klonos** [comment]
Thanx for this! :+1:
